### DISCUSSION TITLE
Custom Noun chunk implementation

### DISCUSSION THREAD

**Discussion by gunalanlakshmanan** [explosion/spaCy#11072] [Help: Coding & Implementations] [answered]
I wanted to change the current noun chunk generator using the custom noun chuck generator. I tried changing the syntax iterators of English Language default.

```
SYNTAX_ITERATORS = {"noun_chunks": noun_chunks_custom}
nlp.Defaults.syntax_iterators = SYNTAX_ITERATORS
```

But when I try to get the noun chuck after changing the defaults it goes to spacy default noun_chunks function in syntax_iterators.py and not the custom function which I have implemented.

Can you please tell me how to implement the custom noun chuck generator function?

Thanks in Advance

**Comment by polm** [comment, marked as answer]
This appears to be a followup to #11059. 

Please note the word is "chunk", not "chuck". 

Can you provide an example of the code that isn't working for you? This works for me.

```
import spacy

from spacy.lang.en import EnglishDefaults

def noun_chunks_custom(doclike):
    print("inside custom function")
    return
    yield

EnglishDefaults.syntax_iterators = {"noun_chunks": noun_chunks_custom}

nlp = spacy.blank("en")

doc = nlp("blah blah")
print(list(doc.noun_chunks))
```

**Reply by gunalanlakshmanan** [reply]
Sorry chuck was a mistake.

I found the problem. I was updating the syntax_iterators after loading the model. After changing the flow by first updating the syntax iterators and then loading the model provided me the right results. 

Thank you.
